File: src/main/java/net/scriptgate/conway/common/Point3D.java
package net.scriptgate.conway.common;

public class Point3D {

    public int x;
    public int y;
    public int z;

    public Point3D() {
    }

    public Point3D(int x, int y, int z) {
        this.x = x;
        this.y = y;
        this.z = z;
    }

    public Point3D(Point3D other) {
        this(other.x, other.y, other.z);
    }

}

File: src/main/java/net/scriptgate/conway/common/Point3D.kt
package net.scriptgate.conway.common

class Point3D {

    var x: Int = 0
    var y: Int = 0
    var z: Int = 0

    constructor() {}

    constructor(x: Int, y: Int, z: Int) {
        this.x = x
        this.y = y
        this.z = z
    }

    constructor(other: Point3D) : this(other.x, other.y, other.z) {}

}
